core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -969,435 +895,571 @@ private void syncLocalToRemote() {
-                  logger.debug(String.format(
-                          Locale.US,
-                          "t='%d': syncLocalToRemote ns=%s documentId=%s local change event "
-                                  + "update description is empty for UPDATE; dropping the event",
-                          logicalT,
-                          nsConfig.getNamespace(),
-                          docConfig.getDocumentId()));
-                  continue;
+                if (unprocessedEventVersionInfo != null) {
+                  final boolean unprocessedEventHasNoVersion =
+                      !unprocessedEventVersionInfo.hasVersion();
+                  final boolean localHasNoVersion = !localVersionInfo.hasVersion();
+
+                  final DocumentVersionInfo.Version unprocessedEventVersion =
+                      unprocessedEventHasNoVersion ? null :
+                          unprocessedEventVersionInfo.getVersion();
+                  final DocumentVersionInfo.Version localVersion =
+                      localHasNoVersion ? null : localVersionInfo.getVersion();
+
+                  if (!unprocessedEventHasNoVersion && !localHasNoVersion) {
+                    final boolean instanceIdMatch = localVersion.getInstanceId()
+                        .equals(unprocessedEventVersion.getInstanceId());
+                    final boolean lastSeenOlderThanRemote =
+                        unprocessedEventVersion.getVersionCounter()
+                            >= localVersion.getVersionCounter();
+
+                    final boolean hasCommittedVersion = instanceIdMatch
+                        && localVersion.getSyncProtocolVersion() == SYNC_PROTOCOL_VERSION
+                        && !lastSeenOlderThanRemote;
+
+                    if (!hasCommittedVersion) {
+                      action = SyncAction.CONFLICT;
+                      message = SyncMessage.VERSION_DIFFERENT_UNPROCESSED_EVENT_MESSAGE;
+                    }
+                  }
                 }
+                // 2. Otherwise, the unprocessed event can be safely dropped and ignored in future
+                // R2L passes. Continue on to checking the operation type.
+              }
 
+              if (action == null) {
+                // iii. Check the operation type
+                switch (localChangeEvent.getOperationType()) {
+                  // 1. INSERT
+                  case INSERT: {
+                    nextVersion =
+                        DocumentVersionInfo.getFreshVersionDocument(
+                            localChangeEvent.getFullDocument());
+
+                    // It's possible that we may insert after a delete happened and we didn't get a
+                    // notification for it. There's nothing we can do about this.
+
+                    // a. Insert document into remote database
+                    try {
+                      remoteColl.insertOne(
```

Comment:
Are you suggesting I store it on the doc config itself rather than as part of the version? That will suppress it being sent to the remote but it breaks the logical connection between the three attributes (guid, version, hash)

Reply:
done